lang/cs/Org.Apache.REEF.IMRU/API/IIMRUClient.cs

```diff
@@ -22,16 +22,17 @@
 namespace Org.Apache.REEF.IMRU.API
 {
     public interface IIMRUClient
-    {        
+    {
         /// <summary>
         /// Submit the given job for execution.
         /// </summary>
         /// <typeparam name="TMapInput">The type of the side information provided to the Map function</typeparam>
         /// <typeparam name="TMapOutput">The return type of the Map function</typeparam>
         /// <typeparam name="TResult">The return type of the computation.</typeparam>
+        /// <typeparam name="TDataHandler">Data Handler type</typeparam>
         /// <param name="jobDefinition">IMRU job definition</param>
         /// <returns>Result of IMRU</returns>
-        IEnumerable<TResult> Submit<TMapInput, TMapOutput, TResult>(IMRUJobDefinition jobDefinition);
+        IEnumerable<TResult> Submit<TMapInput, TMapOutput, TResult, TDataHandler>(IMRUJobDefinition jobDefinition);
```

Comment:
Also, do we really need to introduce these types all the way to the `Submit` method? I struggled with this myself when making the IMRU API. @afchung do you see a way to avoid this pollution of the basic APIs?